src/librustc/ty/layout.rs

```diff
@@ -550,7 +572,7 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
                 let abi = if count != 0 && ty.conservative_is_privately_uninhabited(tcx) {
                     Abi::Uninhabited
                 } else {
-                    Abi::Aggregate { sized: true }
+                    Abi::Aggregate { sized: true, has_vla: (count == 0) }
```

Comment:
Is a check for `repr.c()` not warranted here?

Reply:
Ah, never mind, its an array, which is `repr(C)` by definition.

Reply:
See https://github.com/rust-lang/rust/issues/56877#issuecomment-455684298 about VLA-ness of 0-sized arrays.